workflow/controller/artifact_gc.go

```diff
@@ -151,16 +147,16 @@ func (woc *wfOperationCtx) processArtifactGCStrategy(ctx context.Context, strate
 		}
 		templateName := node.TemplateName
 		if templateName == "" && node.GetTemplateRef() != nil {
-			templateName = node.GetTemplateRef().Name
+			templateName = node.GetTemplateRef().Template
 		}
 		if templateName == "" {
 			return fmt.Errorf("can't process Artifact GC Strategy %s: node %+v has an unnamed template", strategy, node)
 		}
 		template, found := templatesByName[templateName]
 		if !found {
-			template = woc.wf.GetTemplateByName(templateName)
+			template = woc.execWf.GetTemplateByName(templateName)
 			if template == nil {
-				return fmt.Errorf("can't process Artifact GC Strategy %s: template name %q belonging to node %+v not found??", strategy, node.TemplateName, node)
+				return fmt.Errorf("can't process Artifact GC Strategy %s: template name %q belonging to node %+v not found??", strategy, templateName, node)
```

Comment:
note: I confirmed that I don't see any more cases of `woc.wf` used in this file that should really be `woc.execWf`